app/components/base/buttons/button.component.ts

```diff
@@ -33,13 +34,14 @@ export enum SubmitStatus {
     changeDetection: ChangeDetectionStrategy.OnPush,
 
 })
-export class ButtonComponent extends ClickableComponent implements OnChanges {
+export class ButtonComponent extends ClickableComponent implements OnChanges, OnDestroy {
     public SubmitStatus = SubmitStatus;
 
     @Input() public action: ButtonAction;
     @Input() public icon: string;
     @Input() public title: string;
     @Input() public tooltipPosition: string = "above";
+    @Input() public enforcePermission?: boolean;
```

Comment:
can we name this more `permission="write"` as it will give us more flexibility into adding other permissions